gauntlet/packages/gauntlet-solana-contracts/src/lib/contracts.ts

```diff
@@ -36,7 +38,27 @@ const getContractSchema = (name: CONTRACT_LIST, version: string) => {
   return io.readJSON(join(process.cwd(), 'packages/gauntlet-solana-contracts/artifacts/schemas', `${name}`))
 }
 
-const getProgramId = (name: CONTRACT_LIST, version: string) => {
-  const rawPK = io.readJSON(join(process.cwd(), 'packages/gauntlet-solana-contracts/artifacts/programId', `${name}`))
-  return Keypair.fromSecretKey(Uint8Array.from(rawPK))
+const getProgramKeypair = (name: CONTRACT_LIST, version: string): Keypair | undefined => {
+  try {
+    const rawPK = io.readJSON(join(process.cwd(), 'packages/gauntlet-solana-contracts/artifacts/programId', `${name}`))
+    return Keypair.fromSecretKey(Uint8Array.from(rawPK))
+  } catch (e) {
+    logger.warn(`No program id keypair set for program ${name}`)
```

Comment:
Why swallow the error here instead of letting it go to bubble up?

Reply:
Keypair of the program is only necessary for deployments. For interacting we just need the pub key, which we can get from `.env`

Reply:
Yeah, but IMHO we should use it only when we need and in that case, the error should be thrown.

So if we use it (call that fn) when we don't actually need it, and then swallow the err because we didn't actually need it so not interested, that points that we should restructure the code slightly to not actually call it if we don't need it (not deploying).

Minor thing, feel free to keep it as is and move forward for now.

Reply:
That makes sense. Split the function into 2. If a deployment version of the contract is needed, they need ask for it explicitly